src/core/symbol/SymbolTransaction.ts

```diff
@@ -0,0 +1,357 @@
+ */
+export class SymbolTransaction<T extends TransactionBuilder = TransactionBuilder> {
+    /**
+     * The network generation hash used for resolve the transaction hash and the signing data of the transaction.
+     */
+    private readonly generationHash: Uint8Array;
+
+    /**
+     * The serialized payload of a top level simpler or aggregate transaction. Always in sync with the catbuffer builder.
+     *
+     * This field changes when signing and cosigning the transaction.
+     *
+     */
+    private _payload: Uint8Array;
+    /**
+     * The catbuffer builder of a top level simpler or aggregate transaction. Always in sync with the serialized payload.
+     *
+     * This field changes when signing and cosigning the transaction.
+     */
+    private _builder: T;
+
+    /**
+     * Internal constructor, please use the SymbolTransactionFactory
+     * @param network - the network
+     * @param builder - the builder, in-sync with the payload.
+     * @param payload - the serialized payload, in-sync with builder.
+     */
+    public constructor(network: SymbolNetwork, builder: T, payload: Uint8Array) {
+        SymbolTransactionUtils.validateBuilder(builder);
+        this.generationHash = Converter.hexToUint8(network.generationHash);
+        this._payload = payload;
+        this._builder = builder;
+    }
+
+    /**
+     * It creates a transaction from the known payload.
+     * @param network - the network payload
+     * @param payload - the serialized payload, top level simple or aggregate transaction.
+     */
+    public static createFromPayload(network: SymbolNetwork, payload: Uint8Array): SymbolTransaction {
+        return new SymbolTransaction(network, TransactionHelper.loadFromBinary(payload), payload);
+    }
+    /**
+     * It creates a transaction from the builder.
+     * @param network - the network payload
+     * @param builder - the simple or aggregate top level transaction builder.
+     */
+    public static createFromBuilder<T extends TransactionBuilder>(network: SymbolNetwork, builder: T): SymbolTransaction<T> {
+        return new SymbolTransaction(network, builder, builder.serialize());
+    }
+
+    /**
+     * @returns the transaction type of this transaction.
+     */
+    public get transactionType(): number {
+        return this._builder.type;
+    }
+
+    /**
+     * @returns The builder of the top level transaction.
+     */
+    public get builder(): T {
+        return this._builder;
+    }
+
+    /**
+     * @returns the serialized top level transaction
+     */
+    public get payload(): Uint8Array {
+        return this._payload;
+    }
+
+    /**
+     * @returns the size of the transaction.
+     */
+    public get size(): number {
+        return this.builder.size;
+    }
+    /**
+     * @returns the subarray of the paylaod that it needs to be signed by the transaction signer. For information only.
+     */
+    public get signingData(): Uint8Array {
```

Comment:
signingData => signingBytes?

Reply:
KeyPair uses the term `data`

https://github.com/symbol/sdk-typescript/blob/main/src/core/KeyPair.ts#L45

So `signingData` looks better?